lib/daemon_controller.rb

```diff
@@ -778,22 +763,18 @@ def ping_tcp_socket(sockaddr)
   end
 
   def ruby_interpreter
-    rb_config = if defined?(RbConfig)
-      RbConfig::CONFIG
-    else
-      Config::CONFIG
-    end
+    rb_config = defined?(RbConfig)? RbConfig::CONFIG : Config::CONFIG
     File.join(
       rb_config["bindir"],
-      rb_config["RUBY_INSTALL_NAME"]
-    ) + rb_config["EXEEXT"]
+      rb_config.values_at("RUBY_INSTALL_NAME","EXEEXT").join
+    )
   end
 
   def timeoutable(amount, &block)
     Thread.handle_interrupt(Timeout::Error => :never) do
       start_time = monotonic_time
       result = Timeout.timeout(amount, Timeout::Error, &block)
-      [result, [monotonic_time - start_time, 0].max]
+      [result, (monotonic_time - start_time).clamp(0..)]
```

Comment:
`Comparable#clamp` was introduced in Ruby 2.4. We haven't made a decision in Passenger yet about the minimum Ruby version so let's not do this for now.